caffe2-cnn/mnist_nn_cpu.py

```diff
@@ -0,0 +1,181 @@
+      {
+        'name': 'fc3',
+        'type': 'int',
+        'bounds': {
+            'min': 10,
+            'max': 1000,
+        },
+      },
+    ]
+  )
+  return experiment
+
+# Create the model given hyperparameter assignments
+def create_model(assignments):
+  model = cnn.CNNModelHelper(name="mnist_train")
+
+  # Add data to model
+  data_uint8, label = model.TensorProtosDBInput(
+    [], ["data_uint8", "label"], batch_size=64,
+    db=os.path.join(data_folder, 'mnist-train-nchw-leveldb'), db_type='leveldb')
+  data = model.Cast(data_uint8, "data", to=core.DataType.FLOAT)
+  data = model.Scale(data, data, scale=float(1./256))
+  data = model.StopGradient(data, data)
+
+  # Create neural network structure
+  conv1 = model.Conv(data, 'conv1', dim_in=1, dim_out=assignments['conv1_dim'], kernel=5)
+  pool1 = model.MaxPool(conv1, 'pool1', kernel=2, stride=2)
+  conv2 = model.Conv(pool1, 'conv2', dim_in=assignments['conv1_dim'], dim_out=assignments['conv2_dim'], kernel=5)
+  pool2 = model.MaxPool(conv2, 'pool2', kernel=2, stride=2)
+  fc3 = model.FC(pool2, 'fc3', dim_in=assignments['conv2_dim'] * 4 * 4, dim_out=assignments['fc3'])
+  fc3 = model.Relu(fc3, fc3)
+  pred = model.FC(fc3, 'pred', assignments['fc3'], 10)
+  softmax = model.Softmax(pred, 'softmax')
+
+  # Add Training Operators
+  xent = model.LabelCrossEntropy([softmax, label], 'xent')
+  loss = model.AveragedLoss(xent, "loss")
+  accuracy = model.Accuracy([softmax, label], "accuracy")
+  model.AddGradientOperators([loss])
+  ITER = model.Iter("iter")
+  LR = model.LearningRate(
+    ITER, "LR", base_lr=-math.exp(assignments['log_learning_rate']), policy="step", stepsize=1, gamma=0.999 )
+  ONE = model.param_init_net.ConstantFill([], "ONE", shape=[1], value=1.0)
+  for param in model.params:
+    param_grad = model.param_to_grad[param]
+    model.WeightedSum([param, ONE, param_grad, LR], param)
+
+  return model
+
+# Evaluates the model and returns the accuracy
+def train_model(model):
+  workspace.RunNetOnce(model.param_init_net)
+  workspace.CreateNet(model.net)
+  total_iters = 100
+  for i in range(total_iters):
+    workspace.RunNet(model.net.Proto().name)
+
+  return workspace.FetchBlob('accuracy').item()
+
+# Loops through retrieving suggestion, evaluating model, and reporting observation
+def sigopt_optimization_loop(conn, experiment):
+  for _ in range(experiment.observation_budget):
+    # Fetch new suggestion
+    suggestion = conn.experiments(experiment.id).suggestions().create()
+    assignments = suggestion.assignments
+
+    # Test suggestion
+    model = create_model(assignments)
+    accuracy = train_model(model)
+
+    # Report observation
+    conn.experiments(experiment.id).observations().create(
+        suggestion=suggestion.id,
+        value=accuracy
+      )
+
+  # Get most accurate assignments
+  experiment = conn.experiments(experiment.id).fetch()
+  assignments = conn.experiments(experiment.id).best_assignments().fetch()
+
+  print("Best Assignments:")
```

Comment:
We should use `logging` instead of `print`, you can do something like

```python
import logging
logger = logging.getLogger('sigopt')
```

And then you can call `logging.info("message")` instead of print